toucan_connectors/mysql/mysql_connector.py

```diff
@@ -231,25 +235,21 @@ def _connect(
 
         if self.ssl_mode in (SSLMode.VERIFY_CA, SSLMode.VERIFY_IDENTITY):
             ssl_params = self._sanitize_ssl_params()
-            with (
-                NamedTemporaryFile(prefix='ssl_ca') as ssl_ca,
-                NamedTemporaryFile(prefix='ssl_cert') as ssl_cert,
-                NamedTemporaryFile(prefix='ssl_key') as ssl_key,
-            ):
-                # Writing cert contents to temporary files. They're opened in wb+ mode
-                ssl_ca.write(ssl_params['ssl_ca'].encode())
-                ssl_ca.seek(0)
-                ssl_cert.write(ssl_params['ssl_cert'].encode())
-                ssl_cert.seek(0)
-                ssl_key.write(ssl_params['ssl_key'].encode())
-                ssl_key.seek(0)
+            for ssl_opt in ('ssl_ca', 'ssl_key', 'ssl_cert'):
+                ssl_opt_type = NamedTemporaryFile(prefix=ssl_opt)
+                if ssl_opt in ssl_params:
+                    ssl_opt_type.write(ssl_params[ssl_opt].encode())
+                    ssl_opt_type.seek(0)
+
+                    connection_params |= {**connection_params, **{ssl_opt: ssl_opt_type.name}}
 
                 connection_params |= {
-                    'ssl_ca': ssl_ca.name,
-                    'ssl_cert': ssl_cert.name,
-                    'ssl_key': ssl_key.name,
+                    **connection_params,
                     # Verify that the server's hostname matches the CA
-                    'ssl_verify_identity': self.ssl_mode == SSLMode.VERIFY_IDENTITY,
+                    **{
+                        'ssl_verify_identity': self.ssl_mode == SSLMode.VERIFY_IDENTITY
+                        or (getattr(self, 'ssl_key') is None or getattr(self, 'ssl_cert') is None)
+                    },
```

Comment:
This beahviour is not valid: We only want to check if the hostname matches the cert if the mode is VERIFY_IDENTITY